qiita_db/metadata_template.py

```diff
@@ -620,23 +620,37 @@ def __setitem__(self, column, value):
         """
         conn_handler = SQLConnectionHandler()
 
-        exists = conn_handler.execute_fetchone("""
+        # try dynamic tables
+        exists_dynamic = conn_handler.execute_fetchone("""
             SELECT column_name
             FROM information_schema.columns
             WHERE table_name='{0}'
                 AND table_schema='qiita'
                 AND column_name='{1}'""".format(self._dynamic_table, column))
-
-        if exists is None:
+        # try required_sample_info
+        exists_required = conn_handler.execute_fetchone("""
```

Comment:
Oh, I see what's going on here. Please either use `exists` in the SQL to turn it into a bool, or preferably the get_table_cols and `column in table_cols`. Cleaner logic.